### PR TITLE
Azure functions and extensions logger

### PR DESCRIPTION
**Pull request by SeanFeldman**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

- Azure Functions (Isolated Worker/Out-of-Process) support (https://github.com/getsentry/sentry-dotnet/pull/2346)
- Add tag filters to `SentryOptions` (https://github.com/getsentry/sentry-dotnet/pull/2367)

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+```powershell {tabTitle:Package Manager}
+Install-Package Sentry.AzureFunctions.Worker -Version {{@inject packages.version('sentry.dotnet.azurefunctions.worker') }}
+```
+
+```shell {tabTitle:.NET Core CLI}
+dotnet add package Sentry.AzureFunctions.Worker -v {{@inject packages.version('sentry.dotnet.azurefunctions.worker') }}
+```
+
+You can combine this integration with a logging library like `log4net`, `NLog`, or `Serilog`. The logging integration also capture events when an error is logged.
+
```
Comment (lizokm) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:22]:
```suggestion
You can combine this integration with a logging library like `log4net`, `NLog`, or `Serilog`. The logging integration also captures events when an error is logged.

```

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+### Configuring
```
Comment (lizokm) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:23]:
```suggestion
## Configure
```

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+All `ASP.NET Core` configurations are valid here. But one configuration in particular is relevant.
+
+`FlushOnCompletedRequest` ensures all events are flushed out. This is because the general ASP.NET Core hooks for when the process is exiting are not guaranteed to run in a serverless environment. This setting ensures that no event is lost if AWS recycles the process.
```
Comment (lizokm) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:27]:
```suggestion
While all `ASP.NET Core` configurations are valid here, the `FlushOnCompletedRequest` configuration in particular is relevant. That's because general ASP.NET Core hooks for when the process is exiting are not guaranteed to run in a serverless environment and `FlushOnCompletedRequest` flushes all events out. This ensures that no event is lost if AWS recycles the process.
```

**Review by lizokm — APPROVED** [review]
Made some copy suggestions.

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+title: Azure Functions Isolated Worker
+description: "Learn about Sentry's .NET integration with Azure Functions Isolated Worker."
+---
+
+Sentry provides an integration with Azure Functions Isolated Worker through the [Sentry.AzureFunctions.Worker NuGet package](https://www.nuget.org/packages/Sentry.AzureFunctions.Worker).
+All triggers are supported.
+
+## Install
+
+Add the Sentry dependency:
```
[the remark this answers is not in the document]
Comment (SeanFeldman) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx:11]: > Sentry captures data by using an SDK within your application’s runtime. Install the Azure Functions Worker package to add the Sentry dependency:

@lizokm, I'm not sure the sentence conveys the right message. "Sentry captures data using an SDK within your application’s runtime." That's too general. In this specific scenario, Sentry is using Azure Functions Isolated Worker SDK.

Also, "Install the Azure Functions Worker package to add the Sentry dependency:" is misleading as "Azure Functions Worker package" is not a Sentry package but Microsoft. The sentry package would be the `Sentry.AzureFunctions.Worker` package. How about this wording?


> Sentry captures data within your application through the Functions middleware. Install the `Sentry.AzureFunctions.Worker` package to add the Sentry dependency:

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx (hunk 3 above)
Comment (SeanFeldman) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx]: This section shouldn't have been there... copy/pasta problem 🙈
Removed it. Please see the updated version.

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (shanamatthews) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx:11]: Hey @SeanFeldman -

Liza was covering my PR reviews while I was sick. Your wording looks great to me!

**Review by shanamatthews — APPROVED** [review]

File: src/platforms/dotnet/guides/azure-functions-worker/config.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+title: Azure Functions Isolated Worker
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/config.yml:1]:
```suggestion
title: Azure Functions
```

Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/config.yml:1]: Let's leave "Isolated Worker" out of the title, so it fits in the side bar.  We can qualify it in the doc text.  If we ever later add a page for a different model, we can decide how to adjust at that time.

```suggestion
title: Azure Functions
```

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 6]
```diff
@@ -0,0 +1,45 @@
+---
+title: Azure Functions Isolated Worker
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:2]:
```suggestion
title: Azure Functions (Isolated Worker)
```

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+description: "Learn about Sentry's .NET integration with Azure Functions Isolated Worker."
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:3]:
```suggestion
description: "Learn about Sentry's .NET integration with Azure Functions."
```

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+Sentry provides an integration with Azure Functions Isolated Worker through the [Sentry.AzureFunctions.Worker NuGet package](https://www.nuget.org/packages/Sentry.AzureFunctions.Worker).
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:6]:
```suggestion
Sentry provides an integration with Azure Functions through the [Sentry.AzureFunctions.Worker NuGet package](https://www.nuget.org/packages/Sentry.AzureFunctions.Worker).
```

<Note>

This package requires using the Azure Functions ["Isolated Worker Process"](https://learn.microsoft.com/azure/azure-functions/dotnet-isolated-process-guide) execution model.

</Note>

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mattjohnsonpint) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx:11]: Less is more.  This particular line does not need to explain much.

```suggestion

Add the Sentry dependency to your Azure Functions application:
```

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+```powershell {tabTitle:Package Manager}
+Install-Package Sentry.AzureFunctions.Worker -Version {{@inject packages.version('sentry.dotnet.azurefunctions.worker') }}
+```
+
+```shell {tabTitle:.NET Core CLI}
+dotnet add package Sentry.AzureFunctions.Worker -v {{@inject packages.version('sentry.dotnet.azurefunctions.worker') }}
+```
+
+You can combine this integration with a logging library like `log4net`, `NLog`, or `Serilog`. The logging integration also captures events when an error is logged.
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:21]:
```suggestion
This package extends [Sentry.Extensions.Logging](/platforms/dotnet/guides/extensions-logging/). This means that besides the Azure Functions related features, through this package you'll also get access to the `ILogger<T>` integration and also the features available in the main [Sentry](/platforms/dotnet/) SDK.
```

(don't mention log4net, nlog or serilog here)

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+Add the Sentry dependency:
+
+```powershell {tabTitle:Package Manager}
+Install-Package Sentry.AzureFunctions.Worker -Version {{@inject packages.version('sentry.dotnet.azurefunctions.worker') }}
+```
+
+```shell {tabTitle:.NET Core CLI}
+dotnet add package Sentry.AzureFunctions.Worker -v {{@inject packages.version('sentry.dotnet.azurefunctions.worker') }}
+```
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:19]: The .NET CLI command should come first.

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+You can combine this integration with a logging library like `log4net`, `NLog`, or `Serilog`. The logging integration also captures events when an error is logged.
+
+## Configure
+
+Sentry integration with Azure Functions Isolated Worker is done by calling `.UseSentry()` and specifying the options. E.g.:
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:25]:
```suggestion
Sentry integration with Azure Functions (when using the Isolated Worker Process execution mode) is done by calling `.UseSentry()` and specifying the options. E.g.:
```

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+```csharp
+var host = new HostBuilder()
+    .ConfigureFunctionsWorkerDefaults((host, builder) =>
+    {
+        builder.UseSentry(host, options =>
+        {
+            // o.Dsn = "<DSN>";
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/azure-functions-worker/index.mdx:33]: Let the docs fill in the DSN

```suggestion
            options.Dsn = "___PUBLIC_DSN___";
```

File: src/platforms/dotnet/guides/extensions-logging/index.mdx [hunk 13]
```diff
@@ -150,6 +150,10 @@ Whether or not this integration should initialize the SDK. If you intend to call
 
 A list of filters which are invoked before processing any log message. This allows you to inspect the details of the log entry before they become a `Breadcrumb` or an `Event` with full access to the `Microsoft.Extensions.Logging` data.
 
+#### TagFilters
+
+A list of filters which are invoked before processing any log message to filter out undesired tags.
+
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/extensions-logging/index.mdx:156]: Add an example please.

**Review by mattjohnsonpint — CHANGES_REQUESTED** [review]

**Comment by mattjohnsonpint** [comment]
Also requires: https://github.com/getsentry/sentry-release-registry/pull/112

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx (hunk 8 above)
Comment (SeanFeldman) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx:6]: https://www.martinfowler.com/bliki/TwoHardThings.html

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx (hunk 10 above)
Comment (SeanFeldman) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx:19]: I looked up another article and carried the order from there. Do we still want to have .NET CLI first?

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx (hunk 12 above)
Comment (SeanFeldman) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx:33]: That's nice!

File: src/platforms/dotnet/guides/extensions-logging/index.mdx (hunk 13 above)
Comment (SeanFeldman) [reply src/platforms/dotnet/guides/extensions-logging/index.mdx:166]: Done

File: src/platforms/dotnet/guides/azure-functions-worker/index.mdx (hunk 10 above)
Comment (SeanFeldman) [reply src/platforms/dotnet/guides/azure-functions-worker/index.mdx]: I've swapped the order. If the decision is to keep it consistent with the rest of the sample, we can revert later.

File: src/platforms/dotnet/guides/extensions-logging/index.mdx [hunk 14]
```diff
@@ -150,6 +150,20 @@ Whether or not this integration should initialize the SDK. If you intend to call
...
+A list of filters which are invoked before processing any log message to filter out undesired tags. 
+
+E.g. remove all tags starteing with `PROC_` by registering a tag filter.
+
+```csharp
+SentrySdk.Init(options =>
+{
+  options.Dsn = "___PUBLIC_DSN___";
+  options.TagFilters.Add("PROC_");
+}
```
Comment (mattjohnsonpint) [inline src/platforms/dotnet/guides/extensions-logging/index.mdx:164]: No need to show the whole init over again.
```suggestion
options.TagFilters.Add("PROC_");
```

**Review by mattjohnsonpint — APPROVED** [review]
